qtsass/qtsass.py

```diff
@@ -10,12 +13,107 @@
 logger = logging.getLogger(__name__)
 
 
+class Conformer(object):
+    '''Base class for all Text transformations'''
+
+    def to_css(self, qss):
+        '''Transform some qss to valid scss'''
+        return NotImplemented
+
+    def to_qss(self, css):
+        '''Transform some css to valid qss'''
+        return NotImplemented
+
+
+class NotConformer(Conformer):
+    '''Handle QSS "!" in selectors'''
+
+    def to_css(self, qss):
+        '''Replaces "!" not in selectors with "_qnot_"'''
+
+        return qss.replace(':!', ':_qnot_')
+
+    def to_qss(self, css):
+        '''Replaces "_qnot_" in selectors with "!"'''
+
+        return css.replace(':_qnot_', ':!')
+
+
+conformers = [c() for c in Conformer.__subclasses__() if c is not Conformer]
+
+
+def css_conform(input_str):
+    '''Conform qss to valid scss.
```

Comment:
Somewhat a matter of taste, but the Spyder dev generally seems to agree that putting a linebreak before the summary line of a multi-line docstring is preferable as it is clearer and looks cleaner, and that seems to be the convention used elsewhere in the codebase. E.g.

```
"""
Conform qss to valid scss.

Blah blah blah
"""
```